File: speech-recognition/google-cloud-speech-to-text/GoogleCloudSpeechToText.java
// "input" is a string that holds the path to the input file (a local FLAC file, in this example).
// For US English, set "languageCode" to "en-US".

try (SpeechClient client = SpeechClient.create()) {
    Path inputPath = Paths.get(input);
    byte[] bytes = Files.readAllBytes(inputPath);
    ByteString content = ByteString.copyFrom(bytes);

    RecognitionConfig config = RecognitionConfig.newBuilder()
        .setEncoding(RecognitionConfig.AudioEncoding.FLAC)
        .setLanguageCode(languageCode)
        .build();
    RecognitionAudio audio = RecognitionAudio.newBuilder()
        .setContent(content)
        .build();

    RecognizeResponse response = client.recognize(config, audio);

    List<SpeechRecognitionResult> results = response.getResultsList();
}

File: speech-recognition/google-cloud-speech-to-text/GoogleCloudSpeechToText.kt
// "input" is a string that holds the path to the input file (a local FLAC file, in this example).
// For US English, set "languageCode" to "en-US".

val client = SpeechClient.create()

val bytes = input.readBytes()
val content = ByteString.copyFrom(bytes)

val config = RecognitionConfig.newBuilder()
    .setEncoding(RecognitionConfig.AudioEncoding.FLAC)
    .setLanguageCode(languageCode)
    .build()
val audio = RecognitionAudio.newBuilder()
    .setContent(content)
    .build()

val response = client.recognize(config, audio)

val results = response.resultsList
